packages/core/src/api/application.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable jsdoc/check-param-names */
+
+import type { RequestData, REST } from '@discordjs/rest';
+import { type RESTGetCurrentApplicationResult, Routes } from 'discord-api-types/v10';
+
+export class ApplicationsAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Fetches the application associated with the requesting bot user.
+	 *
+	 * @see {@link https://discord.com/developers/docs/resources/application#get-current-application}
+	 * @param options - The options for editing the application
+	 */
+	public async get({ signal }: Pick<RequestData, 'signal'> = {}) {
```

Comment:
This would make the method more consistent with the [`UsersAPI` method](https://github.com/discordjs/discord.js/blob/f6c7c14c9e28c544050b396a4374fdacb0e25fa4/packages/core/src/api/user.ts#L43C3-L43C3).

```suggestion
	public async getCurrent({ signal }: Pick<RequestData, 'signal'> = {}) {
```